File: maestro-sdk/rn/ios/MaestroRnSdk.m
#import "RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(MaestroRnSdk, NSObject)

RCT_EXTERN_METHOD(setup:(NSString)projectId
                  withResolver:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(mockServerUrl:(NSString)baseUrl
                  withResolver:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: maestro-sdk/rn/ios/MaestroRnSdk.swift
import Foundation
import MaestroSDK

@objc(MaestroRnSdk)
class MaestroRnSdk: NSObject {
    
    @objc(setup:withResolver:withRejecter:)
    func setup(
        projectId: String,
        resolve:RCTPromiseResolveBlock,
        reject:RCTPromiseRejectBlock
    ) -> Void {
        MaestroSdk.setup(projectId: projectId)
        resolve(true)
    }
    
    @objc(mockServerUrl:withResolver:withRejecter:)
    func mockServerUrl(
        baseUrl: String,
        resolve:RCTPromiseResolveBlock,
        reject:RCTPromiseRejectBlock
    ) -> Void {
        resolve(
            MaestroSdk.mockServer().url(baseUrl: baseUrl)
        )
    }
    
}
